src/TestHelper/RequestObjectTestHelper.php

```diff
@@ -0,0 +1,188 @@
+        }
+
+        throw new \RuntimeException('There were no validation errors.');
+    }
+
+    /**
+     * Builds an unvalidated request object. The context property will set and overwrite
+     * any properties on the request object with the supplied values.
+     *
+     * @param string $class
+     * @param string $json
+     * @param mixed[]|null $context
+     *
+     * @return \LoyaltyCorp\RequestHandlers\Request\RequestObjectInterface
+     */
+    public function buildUnvalidatedRequest(
+        string $class,
+        string $json,
+        ?array $context = null
+    ): RequestObjectInterface {
+        /** @var \Symfony\Component\Serializer\SerializerInterface $serializer */
+        $serializer = $this->app->get('requesthandlers_serializer');
+
+        /** @var \LoyaltyCorp\RequestHandlers\Request\RequestObjectInterface $instance */
+        $instance = $serializer->deserialize(
+            $json,
+            $class,
+            'json',
+            [
+                PropertyNormalizer::EXTRA_PARAMETERS => $context ?? []
+            ]
+        );
+
+        return $instance;
+    }
+
+    /**
+     * Builds a validated request object. The context property will set and overwrite
+     * any properties on the request object with the supplied values.
+     *
+     * @param string $class
+     * @param string $json
+     * @param mixed[]|null $context
+     *
+     * @return \LoyaltyCorp\RequestHandlers\Request\RequestObjectInterface
+     *
+     * @throws \LoyaltyCorp\RequestHandlers\TestHelper\Exceptions\ValidationFailedException
+     */
+    public function buildValidatedRequest(
+        string $class,
+        string $json,
+        ?array $context = null
+    ): RequestObjectInterface {
+        $instance = $this->buildUnvalidatedRequest($class, $json, $context);
+        $violations = $this->validateRequest($instance);
+
+        if ($violations->count() > 0) {
+            throw new ValidationFailedException($violations);
+        }
+
+        return $instance;
+    }
+
+    /**
+     * Returns an array of properties and their values when those properties have getters.
+     *
+     * @param \LoyaltyCorp\RequestHandlers\Request\RequestObjectInterface $object
+     *
+     * @return mixed[]
+     */
+    public function getRequestProperties(RequestObjectInterface $object): array
+    {
+        $interfaceMethods = \get_class_methods(RequestObjectInterface::class);
+        $instanceMethods = \get_class_methods($object);
+
+        $methodsToCheck = \array_filter(
+            \array_diff($instanceMethods, $interfaceMethods),
+            static function (string $method): bool {
+                return \strncmp($method, 'get', 3) === 0;
+            }
+        );
+
+        $actual = [];
+        foreach ($methodsToCheck as $method) {
+            $property = \lcfirst(\substr($method, 3));
+
+            $callable = [$object, $method];
+            if (\is_callable($callable) === false) {
```

Comment:
I wonder if with these situations, we should squash the check for this, rather than having untested code.

Reply:
Its code that literally will never run, but I think its better to have it here than suppressions with phpstan.neon (which cant target specific lines)

Reply:
Oh yeah, I forgot they had to go in phpstan.neon (ew).